url85e313f409_edugasser_Ascensor_tgz-pJ8-ascensor_MainJ8/src/ascensor/Main.java

```diff
@@ -0,0 +1,778 @@
+             clock = event_list.getS();   
+         }else{   
+             clock = event_list.getR();    
+         }  
+     }
+    
+     public double mean_of(int[] a){
+         int total=0;
+         for (int i =0; i< a.length; i++)
+         {
+             total += a[i];
+         }
+         return total/a.length;
+     }
+     public double variance_of(int[] valores, int size, double mean)
+     {
+        double acum = 0;
+        int i;
+        for (i=0; i<size; i++) acum += Math.pow((double)(valores[i]-mean),2);
+        return acum/(size-1);
+     }
+     public ArrayList e = new ArrayList<>();
+     
+     public boolean procesar_resultados()
+     {
+        TDistribution t = new TDistribution(TOTAL_TRAZAS);
+        double media = mean_of(mean_waiting_time); /* media de todas las trazas */
+        double varianza = variance_of(mean_waiting_time,TOTAL_TRAZAS,media);
+        double tstudent = 0;
+        
+        if(TOTAL_TRAZAS > 400){
+            tstudent = 1.96;
+        }else{   
+            tstudent = Math.abs(t.inverseCumulativeProbability(0.025));  
+        }
+        double ic = tstudent * Math.sqrt(varianza / TOTAL_TRAZAS);
+        double error = ic / media;
+        
+        if (error <= (0.1/(0.1 + 1))){
+            System.out.println((int)media);
+            //System.out.println("var: " + varianza);
+            //System.out.println("Tstudent: " + tstudent);
+            // System.out.println("intervalo confianza: " + ic);
+            // System.out.println("error relativo: " + error);
+        }
+        
+        return error <= (0.1/(0.1 + 1));
+     }
+     public void traza()
+     {    
+         inicializar_traza();
+         while (number_delayed[traza] < K)
+         {  
+            temporizacion();  
+
+            if (clock == event_list.getA())
+            {
+                llegada_pasajero();           
+            }else if( clock == event_list.getL()){              
+                llegada_ascensor();              
+            }else if (clock == event_list.getS()){
+                salida_ascensor();      
+            }else{
+                fin_reflexion();   
+            }
+            
+         } 
+          
+       mean_waiting_time[traza] = total_delayed[traza]/(number_delayed[traza]-cota_transitorio); 
+         //mean_waiting_time[traza] = total_delayed[traza]/(number_delayed[traza]);
+         //System.out.println(mean_waiting_time[traza]);
+         
+     }
+    public void conseguir_transitorio()
+    {
+        inicializar_colas();
+        for (int j = 1; j <100; j++){
+          K = j;
+          aceptable = false;
+          while (!aceptable)
+          {
+              for (int i = 0; i< TOTAL_TRAZAS; i++)
+              {
+                  traza();
+                  traza++;      
+              }
+              aceptable = procesar_resultados();
+              traza = 0;
+          }
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 694:702, 711:719, 741:749.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.